src/history/repeat_commands/repeat_revision.ts

```diff
@@ -0,0 +1,38 @@
+import { Revision } from "../../collaborative/revisions";
+import {
+  getRepeatedCoreCommands,
+  getRepeatedLocalCommand,
+  repeatableLocalCommandsRegistry,
+  repeatCommandTransformRegistry,
+} from "../../registries/repeat_commands_registry";
+import { CoreCommand, Getters } from "../../types";
+import { CommandTypes, coreTypes, isCoreCommand, LocalCommand } from "../../types/commands";
+
+export function canRepeatRevision(revision: Revision | undefined): boolean {
+  if (!revision || !revision.rootCommand || typeof revision.rootCommand !== "object") {
+    return false;
+  }
+
+  // root is core command
+  if ((coreTypes as Set<CommandTypes>).has(revision.rootCommand.type)) {
+    return repeatCommandTransformRegistry.contains(revision.rootCommand.type);
+  }
+
+  return repeatableLocalCommandsRegistry.contains(revision.rootCommand.type);
+}
+
+export function repeatRevision(
+  revision: Revision,
+  getters: Getters
+): CoreCommand[] | LocalCommand | undefined {
+  const commands = revision.commands;
+  if (!revision.rootCommand || typeof revision.rootCommand !== "object") {
+    return undefined;
+  }
+
+  if (isCoreCommand(revision.rootCommand)) {
+    return getRepeatedCoreCommands(getters, commands);
+  }
```

Comment:
if `rootCommand` is a core command, than `commands` is `[revision.rootCommand]`.
Maybe it's more straight forward to understand/explicit like this? What do you think ?
```suggestion
  if (isCoreCommand(revision.rootCommand)) {
    return getRepeatedCoreCommands(getters, [revision.rootCommand]);
  }
```